design_drafts/hardware_access.md

```diff
@@ -0,0 +1,160 @@
+
+**System**
+A system is meant to be a more complex hardware setup which contains multiple joints and sensors.
+This is mostly used for third-party robotic systems such as robotic arms or industrial setups, which have their own (proprietary) API.
+The implementation of the system hardware resource thus serves as an interface between the hardware and the controller to provide this API.
+```xml
+<ros2_control name="MyComplexRobot" type="system">
+  <hardware>
+    <class>complex_robot_pkg/ComplexRobot</class>
+    ...
+    </hardware>
+    <joint name="joint1">
+      <command_interface>velocity</command_interface>
+      <command_interface>effort</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+    </joint>
+    <joint name="joint2">
+      <command_interface>velocity</command_interface>
+      <command_interface>effort</command_interface>
+      <state_interface>position</state_interface>
+      <state_interface>velocity</state_interface>
+      <state_interface>effort</state_interface>
+    </joint>
+</ros2_control>
+```
+
+## Resource Manager
+The resource manager is responsible for parsing the URDF and instantiating the respective hardware resources.
+It has ownership over the lifetime of these hardware resources and their associated logical joints.
+It serves as the storage backend for the controller manager which can loan resources to the controller.
+The resource manager hereby administrates which controllers claimed which resources.
+If a controller does no longer need access to the claimed resource, that resource is being released and offered for other controllers to be claimed.
+
+The resource manager internally maintains a mapping of each individual hardware resources and their interfaces.
+This mapping can be indexed through a simple `<logical_component>/<interface_name>` lookup.
+As mentioned previously it abstracts the individual hardware resources from their logical components, such that a user (or controller in that sense) does not have to know which hardware is responsible for commanding a joint.
+
+In the examples above, the actuator command interfaces are being mapped to `base_link/position`, the state interfaces equivalently to `base_link/position`.
+Likewise for the sensor, their interfaces are being mapped to `my_sensor/roll`, `my_sensor/pitch`, `my_sensor/yaw`.
+
+## Controller Interface
+Once the system is bootstrapped and a controller is loaded, it can claim logical components and access their interfaces.
+
+**Generic Access**
+A controller has the chance to access a single interface value by directly indexing the resource manager for the respective key, such as `joint1/effort`, which allows to set effort values during the controllers execution.
+```c++
+void MyController::init(ResourceManager * resource_manager)
```

Comment:
A code snippet here is unfortunate as now we can start debating implementation approaches. One could argue that this is only a reference, the problem with that is they are sticky and then may be used later as justification. Please forgive me :D 

No raw pointers please. 
A simple change:
```suggestion
void MyController::init(... resource_manager)
```